examples/gno.land/r/demo/upgradable/admin/logic.gno

```diff
@@ -0,0 +1,21 @@
+package admin
+
+import (
+	"std"
+)
+
+type Logic interface {
+	ReadCounter() uint64
+	UpdateCounter()
+
+	DebugPrevRealm() string
+}
+
+func RegisterLogic(l Logic, setStore func(Store)) {
```

Comment:
is there some way to deprovision the old logic? I think there are some similar logic in x/manfred_upgrade_patterns/upgrade_c.

Reply:
When a new logic is registered, a store struct is created simultaneously revoking the previous one(only the newest store is recognized as valid). So if someone call a deprecated version of logic, it will not be able to modify the store with its reference.

Reply:
Yes, I was just thinking about a mechanism to ensure that deprecated logic won't be misused. Currently, v1.Init() can be reused again. However, this doesn't affect the current implementation logic. Thank you for your explanation.